pkg/paymentmethod/client.go

```diff
@@ -22,31 +19,19 @@ type Client interface {
 
 // client is the implementation of Client.
 type client struct {
-	config *config.Config
+	cfg *config.Config
 }
 
 // NewClient returns a new Payment Methods API Client.
-func NewClient(c *config.Config) Client {
-	return &client{
-		config: c,
-	}
+func NewClient(cfg *config.Config) Client {
+	return &client{cfg}
 }
 
 func (c *client) List(ctx context.Context) ([]Response, error) {
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
-	if err != nil {
-		return nil, fmt.Errorf("error creating request: %w", err)
-	}
-
-	res, err := httpclient.Send(ctx, c.config, req)
+	res, err := httpclient.Get[[]Response](ctx, c.cfg, path)
 	if err != nil {
 		return nil, err
 	}
 
-	var formatted []Response
-	if err := json.Unmarshal(res, &formatted); err != nil {
-		return nil, err
-	}
-
-	return formatted, nil
+	return *res, nil
```

Comment:
um slice já é um ponteiro
```suggestion
	return res, nil
```

Reply:
Estou pegando o valor do ponteiro, não definindo um para lista